workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/DialogSession.kt

```diff
@@ -68,7 +74,18 @@ internal class DialogSession(
         }
 
         override fun dispatchKeyEvent(event: KeyEvent): Boolean {
-          return !allowEvents || realWindowCallback.dispatchKeyEvent(event)
+          // Consume all events if we've been told to do so.
+          if (!allowEvents) return true
+
+          // If there is an onBackPressed handler invoke it instead of allowing
+          // the normal machinery to call Dialog.onBackPressed.
+          if (event.isBackPress) {
+            holder.onBackPressed?.invoke()
+            return true
```

Comment:
I thought if the `onBackPressed` is not specified we don't want to block/"handle" back and _should_ actually let it fall through (as the comment below suggests).

Reply:
Wow, good catch. You're absolutely right, and I don't understand how this didn't break `AlertOverlayDialogFactory` -- must be missing tests.

Reply:
Yup, totally broke it.

Reply:
Fixed, updated the comment you mentioned, and added new `canGoBackFromAlert()` test.

```kotlin
          // If there is an onBackPressed handler invoke it instead of allowing
          // the normal machinery to call Dialog.onBackPressed.
          if (event.isBackPress) holder.onBackPressed?.let { onBackPressed ->
            onBackPressed.invoke()
            return true
          }

          // Allow the usual handling, including the usual call to Dialog.onBackPressed.
          return realWindowCallback.dispatchKeyEvent(event)
```